src/panel/ServicePanel.jsx

```diff
@@ -1,69 +1,27 @@
 /* global _ */
 import React from 'react';
-import { Fragment } from 'react';
-import Panel from 'src/components/ui/Panel';
-import CategoryService from 'src/adapters/category_service';
-import CategoryList from 'src/components/CategoryList';
-import nconf from '@qwant/nconf-getter';
+import ServicePanelDesktop from './ServicePanelDesktop';
+import ServicePanelMobile from './ServicePanelMobile';
+import { isMobileDevice } from '../libs/device';
+
 
 class ServicePanel extends React.Component {
   render() {
-    return <Panel
-      resizable
-      title={_('Qwant Maps services', 'service panel')}
-      minimizedTitle={_('Show Qwant Maps services', 'service panel')}
-      className="service_panel"
-      white
-    >
-      <CategoryList className="service_panel__categories" />
-
-      <hr/>
-
-      {
-        nconf.get().events.enabled &&
-        <Fragment>
-          <div className="service_panel__events">
-            <h3>{_('Good plans', 'service panel')}</h3>
-            {
-              CategoryService.getEvents().map(item =>
-                <button className="service_panel__event" type="button" key={item.name}
-                  onClick={() => { window.app.navigateTo(`/events/?type=${item.name}`); }}>
-                  <div className="service_panel__event__icon"
-                    style={{ background: item.backgroundColor }}>
-                    <span className={`icon icon-${item.iconName}`}/>
-                  </div>
-                  <div className="service_panel__category__title">{item.label}</div>
-                </button>
-              )
-            }
-          </div>
-          <hr/>
-        </Fragment>
-      }
-
-      <div className="service_panel__actions">
-
-        {
-          nconf.get().direction.enabled && <button
-            onClick={() => { window.app.navigateTo('/routes/'); }}
-            className="service_panel__action service_panel__item__direction">
-            <div className="service_panel__action__icon">
-              <span className="icon-corner-up-right"/>
-            </div>
-            <div className="service_panel__action__title">{_('Directions', 'service panel')}</div>
-          </button>
-        }
-
-        <button className="service_panel__action service_panel__item__fav"
-          onClick={() => { window.app.navigateTo('/favs'); }}>
-          <div className="service_panel__action__icon">
-            <span className="icon-icon_star"/>
-          </div>
-          <div className="service_panel__action__title">{ _('Favorites', 'service panel')}</div>
-        </button>
-
-      </div>
-    </Panel>;
+    if (isMobileDevice()) {
```

Comment:
If you use this function for that, it won't re-render dynamically if the device changes (like it could happen for a tablet changing orientation).
For that, I introduced a React context in this PR https://github.com/QwantResearch/erdapfel/pull/520, which will update the component automatically if the device changes.
You can find how it's used:
 - here for a class component: https://github.com/QwantResearch/erdapfel/blob/master/src/panel/direction/DirectionPanel.jsx#L329
 - here for a functional component: https://github.com/QwantResearch/erdapfel/blob/master/src/panel/direction/Route.jsx#L11. It may be the occasion to convert your file to a functional component.

Reply:
> It may be the occasion to convert your file to a functional component.

I don't think so: in the near future desktop and mobile HTML/CSS will be reunited and won't need 2 separate subfiles like they do now, so there's no need to lose time on that, no?

Reply:
As you wish. In this case there is no real consequence except code clarity. I personnaly think `useContext` is a simpler thing to use than a context consumer with is child function, but we may have different opinions.
As long as you're at ease with the code.